noxfile.py

```diff
@@ -1,9 +1,47 @@
+from pathlib import Path
+
 import nox
 
-@nox.session(python=["3.11"]) # The python version should match the readthedocs configuration.
+requirements_files = list(
+    {path.name.replace(".in", "") for path in Path(".").glob("*in")}
+)
+
+
+@nox.session(name="pip-compile", python=["3.11"])
+@nox.parametrize(["req"], requirements_files, requirements_files)
+def pip_compile(session: nox.Session, req: str):
+    """Generate lock files from input files or upgrade packages in lock files."""
+    # fmt: off
+    session.install(
+      "-r", "pip-tools.in",
+      "-c", "pip-tools.txt",
+    )
+    # fmt: on
+
+    # Use --upgrade by default unless a user passes -P.
+    upgrade_related_cli_flags = "-P", "--upgrade-package", "--no-upgrade"
+    has_upgrade_related_cli_flags = any(
+        arg.startswith(upgrade_related_cli_flags) for arg in session.posargs
+    )
+    injected_extra_cli_args = () if has_upgrade_related_cli_flags else ("--upgrade",)
+
+    session.run(
+        "pip-compile",
+        "--output-file",
```

Comment:
I would put these in a separate `requirements` directory to avoid polluting the repository root, but that's a subjective point.

Reply:
Yes, I usually suggest that too.

Reply:
I've moved the files into a `requirements` directory. Now I'm wondering if it would be better to name the directory `tests` for consistency with other repos. Any opinions on that?

Reply:
No, there's no tests in this repo.

Reply:
Yeah, I agree. I kept the `tests` directory as a place to store requirements file in the ansible-documentation repo for historical reasons, but I'd prefer not to copy that here.